src/pybind/mgr/dashboard/controllers/docs.py

```diff
@@ -337,7 +337,9 @@ def _gen_spec(cls, all_endpoints=False, base_url="", offline=False):
         if all_endpoints:
             base_url = ""
 
-        host = cherrypy.request.base.split('://', 1)[1] if not offline else 'example.com'
+        host = cherrypy.request.headers.get('Host')
```

Comment:
[X-Forwarded-Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host)?

Reply:
HAProxy doesn't provide config option to set X-Forwarded-Host

Reply:
@epuertat @avanthakkar AFAIK CherryPy is not yet part of [Proxy-protocol ready software](https://www.haproxy.com/blog/haproxy/proxy-protocol/),
so for now let's go for [Forwarder](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) header.

Reply:
Agree on using Forwarded header instead, so that we don't have to pass & fetch multiple headers. Thanks @alfonsomthd . I've updated the changes accordingly and here's the updated [haproxy.cfg](https://pastebin.com/dSziPeCT)